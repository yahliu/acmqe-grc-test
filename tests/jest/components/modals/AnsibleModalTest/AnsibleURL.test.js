/* Copyright (c) 2020 Red Hat, Inc. */
/* Copyright Contributors to the Open Cluster Management project */

'use strict'

import React from 'react'
import { renderAnsibleURL } from '../../../../../src-web/components/modals/AnsibleAutomationModal/AnsibleURL'
import { AcmIcon, AcmIconVariant } from '@open-cluster-management/ui-components'

describe('render AnsibleURL component', () => {
  it('AnsibleURL link renders as empty text', () => {
    const component = renderAnsibleURL('testAnsibleURL', '', '/multicloud/credentials', 3)
    expect(component).toMatchSnapshot()
  })

  it('AnsibleURL link renders as empty text', () => {
    const component = renderAnsibleURL(
      'testAnsibleURL', '', '/multicloud/credentials', 3, true, null
      )
    expect(component).toMatchSnapshot()
  })

  it('AnsibleURL link renders as "-" text', () => {
    const component = renderAnsibleURL('testAnsibleURL', '-', '/multicloud/credentials', 3)
    expect(component).toMatchSnapshot()
  })

  it('AnsibleURL link renders as "-" text', () => {
    const component = renderAnsibleURL(
      'testAnsibleURL', '-', '/multicloud/credentials', 3, false,
      <AcmIcon icon={AcmIconVariant.openNewTab} />)
    expect(component).toMatchSnapshot()
  })

  it('AnsibleURL link renders as normal text with truncate', () => {
    const component = renderAnsibleURL('testAnsibleURL', 'This is an ansible url', '/multicloud/credentials', 3)
    expect(component).toMatchSnapshot()
  })

  it('AnsibleURL link renders as normal text with truncate', () => {
    const component = renderAnsibleURL(
      'testAnsibleURL', 'This is an ansible url', '/multicloud/credentials', 3,false,
      <AcmIcon icon={AcmIconVariant.openNewTab} />
      )
    expect(component).toMatchSnapshot()
  })

  it('AnsibleURL link renders as normal text without truncate', () => {
    const component = renderAnsibleURL('testAnsibleURL', 'This is an ansible url',  '/multicloud/credentials')
    expect(component).toMatchSnapshot()
  })

  it('AnsibleURL link renders as normal text without truncate', () => {
    const component = renderAnsibleURL(
      'testAnsibleURL', 'This is an ansible url',  '/multicloud/credentials', null, false,
      <AcmIcon icon={AcmIconVariant.openNewTab} />
      )
    expect(component).toMatchSnapshot()
  })
})
