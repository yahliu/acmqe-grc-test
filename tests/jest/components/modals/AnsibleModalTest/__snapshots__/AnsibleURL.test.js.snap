// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render AnsibleURL component AnsibleURL link renders as "-" text 1`] = `"-"`;

exports[`render AnsibleURL component AnsibleURL link renders as "-" text 2`] = `"-"`;

exports[`render AnsibleURL component AnsibleURL link renders as empty text 1`] = `
<AcmLaunchLink
  links={
    Array [
      Object {
        "href": "/multicloud/credentials",
        "icon": undefined,
        "id": "testAnsibleURL",
        "noIcon": undefined,
        "text": <TruncateText
          maxCharacters={3}
          text=""
        />,
      },
    ]
  }
/>
`;

exports[`render AnsibleURL component AnsibleURL link renders as empty text 2`] = `
<AcmLaunchLink
  links={
    Array [
      Object {
        "href": "/multicloud/credentials",
        "icon": null,
        "id": "testAnsibleURL",
        "noIcon": true,
        "text": <TruncateText
          maxCharacters={3}
          text=""
        />,
      },
    ]
  }
/>
`;

exports[`render AnsibleURL component AnsibleURL link renders as normal text with truncate 1`] = `
<AcmLaunchLink
  links={
    Array [
      Object {
        "href": "/multicloud/credentials",
        "icon": undefined,
        "id": "testAnsibleURL",
        "noIcon": undefined,
        "text": <TruncateText
          maxCharacters={3}
          text="This is an ansible url"
        />,
      },
    ]
  }
/>
`;

exports[`render AnsibleURL component AnsibleURL link renders as normal text with truncate 2`] = `
<AcmLaunchLink
  links={
    Array [
      Object {
        "href": "/multicloud/credentials",
        "icon": <AcmIcon
          icon="opennewtab"
        />,
        "id": "testAnsibleURL",
        "noIcon": false,
        "text": <TruncateText
          maxCharacters={3}
          text="This is an ansible url"
        />,
      },
    ]
  }
/>
`;

exports[`render AnsibleURL component AnsibleURL link renders as normal text without truncate 1`] = `
<AcmLaunchLink
  links={
    Array [
      Object {
        "href": "/multicloud/credentials",
        "icon": undefined,
        "id": "testAnsibleURL",
        "noIcon": undefined,
        "text": "This is an ansible url",
      },
    ]
  }
/>
`;

exports[`render AnsibleURL component AnsibleURL link renders as normal text without truncate 2`] = `
<AcmLaunchLink
  links={
    Array [
      Object {
        "href": "/multicloud/credentials",
        "icon": <AcmIcon
          icon="opennewtab"
        />,
        "id": "testAnsibleURL",
        "noIcon": false,
        "text": "This is an ansible url",
      },
    ]
  }
/>
`;
